rust/automerge-c/src/obj.rs

```diff
@@ -142,20 +142,35 @@ pub unsafe extern "C" fn AMobjIdIndex(obj_id: *const AMobjId) -> usize {
 /// \brief The type of an object value.
 #[repr(u8)]
 pub enum AMobjType {
+    /// An unknown type.
+    /// \note This tag is unalphabetized to evaluate as false.
+    Unknown = 0,
     /// A list.
-    List = 1,
+    List,
     /// A key-value map.
     Map,
     /// A list of Unicode graphemes.
     Text,
 }
 
+impl From<am::ObjType> for AMobjType {
+    fn from(o: am::ObjType) -> Self {
+        match o {
+            am::ObjType::Map => AMobjType::Map,
+            am::ObjType::List => AMobjType::List,
+            am::ObjType::Table => todo!(),
+            am::ObjType::Text => AMobjType::Text,
+        }
+    }
+}
+
 impl From<AMobjType> for am::ObjType {
     fn from(o: AMobjType) -> Self {
         match o {
             AMobjType::Map => am::ObjType::Map,
             AMobjType::List => am::ObjType::List,
             AMobjType::Text => am::ObjType::Text,
+            AMobjType::Unknown => todo!(),
```

Comment:
Removing the `AMObjType::Uknown` as mentioned earlier would also fix this todo :slightly_smiling_face: